### PR TITLE
JS bindings for loading model and shortlist files as bytes

### PR DESCRIPTION
**Pull request by abhi-agg**
The PR fixes https://github.com/browsermt/bergamot-translator/issues/75

1. Adds JS bindings for `AlignedMemory` class and modifies bindings of `TranslationModel` class to use `AlignedMemory`
2. Modifies wasm test page to use the new bindings
3. Updated README with appropriate instructions

### CODE REVIEW

**Review by andrenatal — DISMISSED** [review]

File: wasm/bindings/TranslationModelBindings.cpp [hunk 1]
```diff
@@ -10,10 +10,30 @@
...
+    .function("size", &marian::bergamot::AlignedMemory::size)
+	  .function("getByteArrayView", &getByteArrayView)
+    ;
+}
+
+TranslationModel* TranslationModelFactory(const std::string &config,
+                                          marian::bergamot::AlignedMemory* modelMemory,
+                                          marian::bergamot::AlignedMemory* shortlistMemory) {
+  if (modelMemory == nullptr || shortlistMemory == nullptr) {
+    return nullptr;
```
Comment (XapaJIaMnu) [inline wasm/bindings/TranslationModelBindings.cpp:29]: Do you ever check for that nullptr down the line? You should have a graceful failure

File: wasm/test_page/bergamot.html [hunk 2]
```diff
@@ -93,22 +111,53 @@
...
+      log(`${languagePair} file download took ${(Date.now() - start) / 1000} secs`);
+
+      // Construct AlignedMemory objects with downloaded buffers
+      var alignedModelMemory = constructAlignedMemoryFromBuffer(modelBuffer, 256);
+      var alignedShortlistMemory = constructAlignedMemoryFromBuffer(shortListBuffer, 64);
+
+      // Instantiate the TranslationModel
+      if (translationModel) translationModel.delete();
+      console.debug("Creating TranslationModel with config:", modelConfigWithoutModelAndShortList);
+      translationModel = new Module.TranslationModel(modelConfigWithoutModelAndShortList, alignedModelMemory, alignedShortlistMemory);
```
Comment (XapaJIaMnu) [inline wasm/test_page/bergamot.html:157]: This could be null here

**Review by XapaJIaMnu — CHANGES_REQUESTED** [review]
Gracefully fail when model construction fails.

File: wasm/bindings/TranslationModelBindings.cpp (hunk 1 above)
Comment (abhi-agg) [reply wasm/bindings/TranslationModelBindings.cpp:29]: We don't expect these to be nullptr. I kept it just to be cautious. I will remove it from here. 👍

File: wasm/test_page/bergamot.html (hunk 2 above)
Comment (abhi-agg) [reply wasm/test_page/bergamot.html:157]: The object returned will never be null once I fix the comment above.

**Comment by XapaJIaMnu** [comment]
I think it's good to have a check for failure. I'm not sure what happens in your pipeline when the download fails. I think I saw some code that covers this case

**Comment by abhi-agg** [comment]
> I think it's good to have a check for failure. I'm not sure what happens in your pipeline when the download fails. I think I saw some code that covers this case

You are right about both handling the download failure and a check in place in the test page for this 👍

**Review by XapaJIaMnu — APPROVED** [review]

**Review by motin — APPROVED** [review]
Approving without testing the changes (will be done once it is possible).

**Pull request merged**